vignettes/NSSP_nowcast.Rmd

```diff
@@ -0,0 +1,539 @@
+We will create a unique event ID using the event number and the patient ID in the `C_Processed_BioSense_ID` column.
+```{r}
+lastcol_ts <- tail(colnames(select(
+  syn_nssp_time_stamps_wide,
+  matches("DischargeDiagnosisMDTUpdates")
+)), 1)
+syn_nssp_time_stamps_long <- syn_nssp_time_stamps_wide |>
+  pivot_longer(
+    cols = starts_with("DischargeDiagnosisMDTUpdates"),
+    names_to = "column_name",
+    values_to = "time_stamp",
+    values_drop_na = FALSE
+  ) |>
+  mutate(
+    event_id = paste(
+      C_Processed_BioSense_ID,
+      parse_number(as.character(column_name))
+    )
+  )
+```
+
+Perform the same procedure for the diagnoses dataset.
+```{r}
+lastcol_diag <- tail(colnames(select(
+  syn_nssp_diagnoses_wide,
+  matches("DischargeDiagnosisUpdates")
+)), 1)
+syn_nssp_diagnoses_long <- syn_nssp_diagnoses_wide |>
+  pivot_longer(
+    cols = starts_with("DischargeDiagnosisUpdates"),
+    names_to = "column_name",
+    values_to = "diagnoses_codes",
+    values_drop_na = FALSE
+  ) |>
+  mutate(
+    event_id = paste(
+      C_Processed_BioSense_ID,
+      parse_number(as.character(column_name))
+    )
+  )
+```
+
+Next, we will clean up the time stamps in the data so that the `time_stamp` column is formatted as `%Y-%m-%d %H:%M:%S`, and then we will filter out an events that are not present (updates are NAs).
+```{r}
+syn_nssp_time_stamps_long <-
+  syn_nssp_time_stamps_long |>
+  mutate(time_stamp = as.POSIXct(
+    str_remove_all(
+      str_remove(time_stamp, ".*\\}"),
+      "[|;]+"
+    ),
+    format = "%Y-%m-%d %H:%M:%S",
+    tz = "UTC"
+  )) |>
+  drop_na(time_stamp)
+```
+
+Clean up the diagnoses codes and remove the empty updates from the diagnoses dataset.
+For these, we want to keep the semi-colons and just remove the numbers since
+this information is stored in the event ID.
+We will only use the event ID and the diagnoses codes, as this will be merged
+back into the time stamped dataset.
+```{r}
+syn_nssp_diagnoses_long <-
+  syn_nssp_diagnoses_long |>
+  mutate(diagnoses_codes = str_remove(diagnoses_codes, ".*\\}")) |>
+  filter(nzchar(diagnoses_codes)) |>
+  drop_na() |>
+  select(event_id, diagnoses_codes)
+```
+
+
+Merge together the time stamps of events and the diagnoses codes.
+Filter to remove empty updates.
+```{r}
+nssp_merged <- merge(syn_nssp_time_stamps_long,
+  syn_nssp_diagnoses_long,
+  by = "event_id"
+) |>
+  filter(diagnoses_codes != ";;|") # drops empty updates
+```
+Now we have a dataframe where each row is an event, with the patient's visit start date (`C_Visit_date_Time`), the patient ID (`C_Processed_BioSense_ID`), the diagnoses code at the event (`diagnoses_code`), and the time stamp of the event (`time_stamp`).
+
+Next we will add a column for the time from arrival to each updated diagnosis, in days.
+```{r}
+nssp_updates <- nssp_merged |>
+  mutate(arrival_to_update_delay = as.numeric(difftime(
+    as.POSIXct(time_stamp), as.POSIXct(C_Visit_Date_Time),
```

Comment:
I would suggest on pass doing a variable transform to posixct and then using that rather than having to keep writing this as it makes the code long and boring to read